x-pack/filebeat/magefile.go

```diff
@@ -188,3 +193,58 @@ func PythonIntegTest(ctx context.Context) error {
 		return devtools.PythonNoseTest(args)
 	}, "GENERATE", "TESTING_FILEBEAT_MODULES", "TESTING_FILEBEAT_FILESETS")
 }
+
+// CreateModule creates a new Filebeat module.
+// Use MODULE=module to specify the name of the new module
+func CreateModule() error {
+	targetModule := os.Getenv("MODULE")
+	if targetModule == "" {
+		return fmt.Errorf("You must specify the module: MODULE=name mage createModule")
+	}
+	xFilebeatDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	filebeatDir := path.Join(path.Dir(path.Dir(xFilebeatDir)), "filebeat")
+
+	return module.Generate(targetModule, xFilebeatDir, filebeatDir)
+}
+
+// CreateFileset creates a new fileset for an existing Filebeat module.
+// Use MODULE=module to specify the name of the existing module
+// Use FILESET=fileset to specify the name of the new fileset
+func CreateFileset() error {
+	targetModule := os.Getenv("MODULE")
+	targetFileset := os.Getenv("FILESET")
+
+	if targetModule == "" || targetFileset == "" {
+		return fmt.Errorf("You must specify the module and fileset: MODULE=module FILESET=fileset createFileset")
+	}
+
+	xFilebeatDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	filebeatDir := path.Join(path.Dir(path.Dir(xFilebeatDir)), "filebeat")
```

Comment:
I think you want https://godoc.org/github.com/elastic/beats/dev-tools/mage#OSSBeatDir.